lib/openshift/resource.rb

```diff
@@ -49,6 +49,14 @@ def exists?(user: nil, result: {}, quiet: false)
       elsif result[:response] =~ /Unable to connect to the server/
         logger.info(result[:response])
         return false
+      # Adding below because when in Single node cluster node restarted then SNO not have other nodes to communicate which cause connection timeout issue.
+      # And to prevent premature exit as describe in OCPQE-8043.
+      no_of_master = env.nodes.length
+      cluster_host = env.api_hostname
+      elsif result[:response] =~ /connection to the server \S+.qe.devcluster.openshift.com:6443 was refused - did you specify the right host or port/ && no_of_master == 1
```

Comment:
Are we certain all clusters are using a base domain as `qe.devcluster.openshift.com` and port `6443`?. This may not be the case when the cluster is created on Baremetal using a different configuration.